monitor/ConsoleReporter.js

```diff
@@ -72,16 +72,20 @@ define(function(require) {
 				// IE8 has console.log and JSON, so we can make a
 				// reasonably useful warn() from those.
 				// Credit to webpro (https://github.com/webpro) for this idea
-				if (typeof localConsole.log ==='function'
-					&& typeof JSON !== 'undefined') {
-					log = warn = function (x) {
-						if(typeof x !== 'string') {
+				// typeof localConsole.log will return 'object' in IE8, so can't test it with === 'function'
+				// Since this is more of a corner case for IE8, I'm ok to check it with !== 'undefined' to reduce complexity
+				if (typeof localConsole.log !== 'undefined' && typeof JSON !== 'undefined') {
```

Comment:
@briancavalier I'm sorry this is my own experience, I don't have a link and I have found no traces about this issue on the Internet. I recall that this happened in certain debugging conditions: I had IE11 in IE8 mode with the console tab not opened initially, and there was some navigation that likely broke the connection between the webpage and the console object (like the closured console instance got destroyed internally and was not substituted by the new one). Maybe this is an issue in the Developer Tools because in IE the `console` object is provided by them. But I saw similar `Access denied` (COM-related) issues for `window` and `location` objects in production; they popped out if references to `window` and `location` were stored in a closure and later accessed after the `document.domain` changed. So I cannot guarantee this won't happen with the `console` object because it's COM-based, too, thus fragile. It is important to make these helper objects cope with any error happening inside of them not to break the surrounding code.